test/stat_tracker_test.rb

```diff
@@ -48,6 +48,80 @@ def test_it_can_get_total_tied_games
+    exclude = "away"
+
+    assert_equal total_home_games_per_team, @@stat_tracker.total_games_per_team(exclude)
+
+    total_away_games_per_team = {3=>5, 6=>4, 5=>4, 17=>1, 16=>3, 14=>3, 28=>5, 54=>3, 24=>2}
+    exclude = "home"
+
+    assert_equal total_away_games_per_team, @@stat_tracker.total_games_per_team(exclude)
+  end
+
+  def test_it_can_get_total_goals_per_team
+    expected = {3=>17, 6=>24, 5=>7, 17=>1, 16=>15, 14=>8, 28=>24, 54=>16, 24=>4}
+
+    assert_equal expected, @@stat_tracker.total_goals_per_team
+  end
+
+  def test_it_can_exclude_hoa_goals_from_total_goals_per_team
+    expected_home_goals_per_team = {6=>12, 3=>8, 5=>3, 16=>8, 14=>4, 54=>11, 28=>13, 24=>2}
+    exclude = "away"
+
+    assert_equal expected_home_goals_per_team, @@stat_tracker.total_goals_per_team(exclude)
+
+    expected_away_goals_per_team = {3=>9, 6=>12, 5=>4, 17=>1, 16=>7, 14=>4, 28=>11, 54=>5, 24=>2}
+    exclude = "home"
+
+    assert_equal expected_away_goals_per_team, @@stat_tracker.total_goals_per_team(exclude)
+  end
+
+  def test_it_can_get_average_goals_per_game_per_team
+    team3 = @@stat_tracker.teams.find { |team| team.id == 3 }
+    team5 = @@stat_tracker.teams.find { |team| team.id == 5 }
+    team6 = @@stat_tracker.teams.find { |team| team.id == 6 }
+    team14 = @@stat_tracker.teams.find { |team| team.id == 14 }
+    team16 = @@stat_tracker.teams.find { |team| team.id == 16 }
+    team17 = @@stat_tracker.teams.find { |team| team.id == 17 }
+    team28 = @@stat_tracker.teams.find { |team| team.id == 28 }
+    team24 = @@stat_tracker.teams.find { |team| team.id == 24 }
+    team54 = @@stat_tracker.teams.find { |team| team.id == 54 }
+
+    expected = {team3=>1.89, team6=>2.67, team5=>0.88, team17=>1.0, team16=>2.14, team14=>1.33, team28=>2.4, team54=>2.67, team24=>1.0}
+
+    assert_equal expected, @@stat_tracker.average_goals_per_game_per_team
+  end
+
+  def test_it_can_get_average_goals_per_game_per_team_hoa
+    team3 = @@stat_tracker.teams.find { |team| team.id == 3 }
+    team5 = @@stat_tracker.teams.find { |team| team.id == 5 }
+    team6 = @@stat_tracker.teams.find { |team| team.id == 6 }
+    team14 = @@stat_tracker.teams.find { |team| team.id == 14 }
+    team16 = @@stat_tracker.teams.find { |team| team.id == 16 }
+    team17 = @@stat_tracker.teams.find { |team| team.id == 17 }
+    team28 = @@stat_tracker.teams.find { |team| team.id == 28 }
+    team24 = @@stat_tracker.teams.find { |team| team.id == 24 }
+    team54 = @@stat_tracker.teams.find { |team| team.id == 54 }
+
+    expected_avg_home_goals_p_game_p_team = {team3=>2.0, team6=>2.4, team5=>0.75, team16=>2.0, team14=>1.33, team28=>2.6, team54=>3.67, team24=>1.0}
+    exclude = "away"
+
+    assert_equal expected_avg_home_goals_p_game_p_team, @@stat_tracker.average_goals_per_game_per_team(exclude)
+
+    expected_avg_away_goals_p_game_p_team = {team14=>1.33, team6=>3.0, team3=>1.8, team5=>1.0, team17=>1.0, team28=>2.2, team16=>2.33, team24=>1.0, team54=>1.67}
```

Comment:
I like this a lot. Will be able to pull this in for highest scoring visitor! Thank you!